src/Cli/dotnet/Installer/Windows/Security/AuthentiCode.cs

```diff
@@ -1,190 +1,61 @@
-        /// organizations; <see langword="false"/> otherwise.</returns>
-        internal static bool IsSignedByTrustedOrganization(string path, params string[] organizations)
+        /// <param name="path">The path of the file to verify.</param>
+        /// <param name="cacheOnlyRevocationChecks">Prevents revocation checks over the network when set to <see langword="true"/>.</param>
+        /// <returns>0 if successful.</returns>
+        public static unsafe int IsSigned(string path, bool cacheOnlyRevocationChecks = false)
         {
-            try
-            {
-                IEnumerable<X509Certificate2> certificates = GetCertificates(path);
-
-                return certificates.Any(c => organizations.Any(o => c.Subject.Contains($"O={o}", StringComparison.OrdinalIgnoreCase)));
-            }
-            catch (CryptographicException)
-            {
-                return false;
-            }
-        }
-
-        /// <summary>
-        /// Retrieves the certificates from each signature in the specified file, including nested signatures.
-        /// </summary>
-        /// <param name="path">The path of the file.</param>
-        /// <returns>A collection of one or more certificates.</returns>
-        internal static List<X509Certificate2> GetCertificates(string path)
-        {
-            List<X509Certificate2> certificates = new();
-
-            try
-            {
-                certificates.Add(new X509Certificate2(X509Certificate.CreateFromSignedFile(path)));
-            }
-            catch (CryptographicException)
-            {
-                return certificates;
-            }
-
-            SignedCms cms = CreateSignedCmsFromFile(path);
-
-            foreach (CryptographicAttributeObject attribute in cms.SignerInfos[0].UnsignedAttributes)
-            {
-                if (attribute.Oid.Value.Equals(OidNestedSignature))
+            WINTRUST_FILE_INFO* fileInfo = stackalloc WINTRUST_FILE_INFO[1];
+            WINTRUST_DATA* trustData = stackalloc WINTRUST_DATA[1];
+
+            fixed (char* p = Path.GetFullPath(path))
+            {
+                fileInfo[0].pcwszFilePath = p;
+                fileInfo[0].cbStruct = (uint)sizeof(WINTRUST_FILE_INFO);
+                fileInfo[0].hFile = (HANDLE)IntPtr.Zero;
+                fileInfo[0].pgKnownSubject = null;
+
+                Guid policyGuid = WINTRUST_ACTION_GENERIC_VERIFY_V2;
+
+                trustData[0].cbStruct = (uint)sizeof(WINTRUST_DATA);
+                trustData[0].pPolicyCallbackData = null;
+                trustData[0].pSIPClientData = null;
+                trustData[0].dwUIChoice = WINTRUST_DATA_UICHOICE.WTD_UI_NONE;
+                trustData[0].fdwRevocationChecks = WINTRUST_DATA_REVOCATION_CHECKS.WTD_REVOKE_WHOLECHAIN;
+                trustData[0].dwUnionChoice = WINTRUST_DATA_UNION_CHOICE.WTD_CHOICE_FILE;
+                trustData[0].dwStateAction = WINTRUST_DATA_STATE_ACTION.WTD_STATEACTION_VERIFY;
```

Comment:
[Docs](https://learn.microsoft.com/windows/win32/api/wintrust/ns-wintrust-wintrust_data) say:

Value | Meaning
-- | --
WTD_STATEACTION_IGNORE 0x00000000 | Ignore the hWVTStateData member.
WTD_STATEACTION_VERIFY 0x00000001 | Verify the trust of the object (typically a file) that is specified by the dwUnionChoice member. The hWVTStateData member will receive a handle to the state data. This handle must be freed by specifying the WTD_STATEACTION_CLOSE action in a subsequent call.
WTD_STATEACTION_CLOSE 0x00000002 | Free the hWVTStateData member previously allocated with the WTD_STATEACTION_VERIFY action. This action must be specified for every use of the WTD_STATEACTION_VERIFY action.

[This sample](https://learn.microsoft.com/windows/win32/seccrypto/example-c-program--verifying-the-signature-of-a-pe-file) shows proper use of WTD_STATEACTION_VERIFY + WTD_STATEACTION_CLOSE.

I don't see a call with WTD_STATEACTION_CLOSE anywhere.  Should you use WTD_STATEACTION_IGNORE instead?

Reply:
No, I think I need to just make the release call to free up the handle.